tools/webdriver/webdriver/bidi/modules/permissions.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any, Optional, Mapping, MutableMapping
+
+from ._module import BidiModule, command
+
+
+class Permissions(BidiModule):
+    @command
+    def end(self) -> Mapping[str, Any]:
+        return {}
+
+    @end.result
+    async def _end(self, result: Mapping[str, Any]) -> Any:
+        if self.session.transport:
+            await self.session.transport.wait_closed()
+
+        return result
+
+    @command
+    def set_permission(self,
+                    descriptor: Optional[Mapping[str, Any]] = None,
+                    state: Optional[str] = None,
+                    origin: Optional[str] = None) -> Mapping[str, Any]:
+        params: MutableMapping[str, Any] = {
+            "descriptor": descriptor,
+            "state": state,
+            "origin": origin
+        }
```

Comment:
I'd assume you want to emit not provided parameters, not set them to `None` (`null`).

Reply:
You can make it like this: 
https://github.com/web-platform-tests/wpt/blob/53d5a05d1cd02abe604721e5baf53e3ced7a960e/tools/webdriver/webdriver/bidi/modules/storage.py#L22C1-L44C36

Reply:
I believe undefined should be done using https://github.com/web-platform-tests/wpt/blob/master/webdriver/tests/bidi/browsing_context/set_viewport/viewport.py#L2 Let me add that